src/Entity/Carrinho.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+namespace App\Entity;
+
+use App\Repository\CarrinhoRepository;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\DBAL\Types\Types;
+use Doctrine\ORM\Mapping as ORM;
+
+#[ORM\Entity(repositoryClass: CarrinhoRepository::class)]
+class Carrinho
+{
+    #[ORM\Id]
+    #[ORM\GeneratedValue]
+    #[ORM\Column]
+    private ?int $id = null;
+
+    #[ORM\OneToOne(inversedBy: 'carrinho', cascade: ['persist', 'remove'])]
+    private ?Cliente $cliente_id = null;
+
+    #[ORM\ManyToOne(inversedBy: 'carrinhos')]
+    private ?User $usuario_id = null;
+
+    #[ORM\Column(length: 255)]
+    private ?string $status = null;
+
+    #[ORM\Column]
+    private ?int $valor_total = null;
+
+    #[ORM\Column(type: Types::DATETIME_MUTABLE)]
+    private ?\DateTimeInterface $criado_em = null;
+
+    #[ORM\Column(type: Types::DATETIME_MUTABLE)]
+    private ?\DateTimeInterface $atualizado_em = null;
+
+    #[ORM\Column(type: Types::DATETIME_MUTABLE)]
+    private ?\DateTimeInterface $finalizado_em = null;
+
+    /**
+     * @var Collection<int, Item>
+     */
+    #[ORM\ManyToMany(targetEntity: Item::class, mappedBy: 'carrinho_id')]
+    private Collection $items;
+
+    public function __construct()
+    {
+        $this->items = new ArrayCollection();
+    }
+
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
+    public function getClienteId(): ?Cliente
+    {
+        return $this->cliente_id;
+    }
+
+    public function setClienteId(?Cliente $cliente_id): static
+    {
+        $this->cliente_id = $cliente_id;
+
+        return $this;
+    }
+
+    public function getUsuarioId(): ?User
+    {
+        return $this->usuario_id;
+    }
+
+    public function setUsuarioId(?User $usuario_id): static
```

Comment:
como estamos retornando a instancia vamos usar self aqui em vez de static, vai ser melhor...
static quando for public static function

```suggestion
    public function setUsuarioId(?User $usuario_id): self
```